proof/refine/ARM/Detype_R.thy

```diff
@@ -4533,26 +4552,97 @@ lemma doMachineOp_ksArchState_commute:
     apply clarsimp+
   done
 
+lemma doMachineOp_ksPSpace:
+  "monad_commute \<top> (doMachineOp f) (gets ksPSpace)"
+  apply (clarsimp simp: monad_commute_def doMachineOp_def bind_assoc)
+  apply monad_eq
+  apply (auto simp add: select_f_def)
+  done
+
+lemma doMachineOp_assert_opt:
+  "empty_fail f \<Longrightarrow> monad_commute \<top> (doMachineOp f) (assert_opt m)"
+  apply (clarsimp simp: monad_commute_def doMachineOp_def bind_assoc)
+  apply monad_eq
+  apply (force simp add: select_f_def empty_fail_def)
+  done
+
+lemma doMachineOp_assert:
+  "empty_fail f \<Longrightarrow> monad_commute \<top> (doMachineOp f) (assert P)"
+  apply (clarsimp simp: monad_commute_def doMachineOp_def bind_assoc)
+  apply monad_eq
+  apply (force simp add: select_f_def empty_fail_def)
+  done
+
+lemma doMachineOp_projectKO_pde:
+  "empty_fail f \<Longrightarrow> monad_commute \<top> (doMachineOp f) (projectKO ko :: pde kernel)"
+  apply (clarsimp simp: monad_commute_def doMachineOp_def bind_assoc projectKO_def)
+  apply monad_eq
+  by (force split: option.splits simp: fail_def return_def select_f_def empty_fail_def)
+
+lemma doMachineOp_alignCheck:
```

Comment:
I might discover this later in the review, but are all the doMachineOp lemmas in this file new, or moved?

Reply:
changed versions of moved lemmas (some where not phrased with monad_commute before, but the originals where in the same theoory). I guess we could move them further up, but we also do not seem to need them anywhere else, so I tried to keep noise low.